scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/DatasetService.java

```diff
@@ -0,0 +1,199 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import edu.asu.diging.scriptoocloud.core.data.DataFileRepository;
+import edu.asu.diging.scriptoocloud.core.data.DatasetRepository;
+import edu.asu.diging.scriptoocloud.core.exceptions.DataFileNotFoundException;
+import edu.asu.diging.scriptoocloud.core.exceptions.DatasetNotFoundException;
+import edu.asu.diging.scriptoocloud.core.exceptions.DatasetStorageException;
+import edu.asu.diging.scriptoocloud.core.exceptions.FileSystemStorageException;
+import edu.asu.diging.scriptoocloud.core.model.IDataset;
+import edu.asu.diging.scriptoocloud.core.model.impl.DataFile;
+import edu.asu.diging.scriptoocloud.core.model.impl.Dataset;
+import edu.asu.diging.scriptoocloud.core.service.IDatasetService;
+import edu.asu.diging.simpleusers.core.model.IUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.Locale;
+import java.util.Optional;
+
+/**
+ * The Service class associated with user Datasets.  The class handles the association between
+ * Datasets and DataFiles they contain.
+ */
+@Transactional
```

Comment:
same as above use the `rollbackFor` attribute to rollback any database changes in case of an exception